lib/ansible/modules/network/nxos/nxos_acl_interfaces.py

```diff
@@ -0,0 +1,427 @@
+
+# Using replaced
+
+# Before state:
+# ------------
+# interface Ethernet1/2
+#   ipv6 traffic-filter ACL1v6 in
+# interface Ethernet1/5
+#   ip port access-group PortACL in
+#   ip access-group ACL1v4 out
+#   ipv6 traffic-filter ACL1v6 in
+
+- name: Replace interface configuration with given configuration
+  nxos_acl_interfaces:
+    config:
+      - name: Eth1/5
+        access_groups:
+          - afi: ipv4
+            acls:
+              - name: NewACLv4
+                direction: out
+
+      - name: Ethernet1/3
+        access_groups:
+          - afi: ipv6
+            acls:
+              - name: NewACLv6
+                direction: in
+                port: true
+    state: replaced
+
+# After state:
+# ------------
+# interface Ethernet1/2
+#   ipv6 traffic-filter ACL1v6 in
+# interface Ethernet1/3
+#   ipv6 port traffic-filter NewACLv6 in
+# interface Ethernet1/5
+#   ip access-group NewACLv4 out
+
+# Using overridden
+
+# Before state:
+# ------------
+# interface Ethernet1/2
+#   ipv6 traffic-filter ACL1v6 in
+# interface Ethernet1/5
+#   ip port access-group PortACL in
+#   ip access-group ACL1v4 out
+#   ipv6 traffic-filter ACL1v6 in
+
+- name: Override interface configuration with given configuration
+  nxos_acl_interfaces:
+    config:
+      - name: Ethernet1/3
+        access_groups:
+          - afi: ipv4
+            acls:
+              - name: ACL1v4
+                direction: out
+
+              - name: PortACL
+                port: true
+                direction: in
+          - afi: ipv6
+            acls:
+              - name: NewACLv6
+                direction: in
+                port: true
+    state: overridden
+
+# After state:
+# ------------
+# interface Ethernet1/3
+#   ip access-group ACL1v4 out
+#   ip port access-group PortACL in
+#   ipv6 port traffic-filter NewACLv6 in
+
+# Using deleted
+
+# Before state:
+# -------------
+# interface Ethernet1/2
+#   ipv6 traffic-filter ACL1v6 in
+# interface Ethernet1/5
+#   ip port access-group PortACL in
+#   ip access-group ACL1v4 out
+#   ipv6 traffic-filter ACL1v6 in
+
+- name: Delete ACL configuration on interfaces
+  nxos_acl_interfaces:
+    config:
+      - name: Ethernet1/5
+        access_groups:
+          - afi: ipv6
+
+          - afi: ipv4
+            acls:
+              - name: ACL1v4
+                direction: out
+
+      - name: Ethernet1/2
+# After state:
+# -------------
+# interface Ethernet1/2
+# interface Ethernet1/5
+#   ip port access-group PortACL in
+#   ip access-group ACL1v4 out
+#   ipv6 traffic-filter ACL1v6 in
+
+# Using parsed
+
+# Before state:
+# ------------
+#
+
+- name: Parse given configuration into structured format
+  nxos_acl_interfaces:
+    running_config: |
+      interface Ethernet1/2
+      ipv6 traffic-filter ACL1v6 in
+      interface Ethernet1/5
+      ipv6 traffic-filter ACL1v6 in
+      ip access-group ACL1v4 out
+      ip port access-group PortACL in
+    state: parsed
+
+# After state:
```

Comment:
No Before and after keys for parsed, rendered and gathered state.

Reply:
Okay, I'll remove them. But shouldn't we have 'before' for gathered, as we are gathering what is in the device?